setup.py

```diff
@@ -22,11 +20,18 @@
 else:
     install_requires.append('requests')
 
+def get_version():
+    for line in open('butter_cms/__version__.py', 'r'):
+        if line.startswith('__version__'):
+            return line.split('=')[1].strip().strip("'")
+
+    return RuntimeError('Unable to find version string.')
```

Comment:
Maybe we could do something similar to the `coverage.py`'s approach, i.e. execute the code in the `__version__.py`, instead of manually parsing something that is the Python code? WDYT?

https://github.com/nedbat/coveragepy/blob/master/setup.py#L41-L48

Reply:
True that manually parsing python code might seem cumbersome, but I think it would be more safe to read python code then execute it, mainly if we don’t need such complex versioning. Also, release-please, when used with type [python](https://github.com/googleapis/release-please?tab=readme-ov-file#strategy-language-types-supported) it expects this kind of structure: https://github.com/googleapis/python-storage. It parses and manages version.py and version in it without tagging it like in [readme](https://github.com/ButterCMS/buttercms-python/blob/6e50b6d93ea76e095bc6c7feb525f8e51fa0c1b9/README.md?plain=1#L1).

Reply:
> I think it would be more safe to read python code then execute it

Good point. I'm uncertain, however, if that's not overkill—we own the `__version__.py` file and have control over what it contains, so I don't think there's a potential security hole for anyone if we execute its content? Or did I overlook something? WDYT?

`googleapis/python-storage` does a similar thing, if you would prefer something simpler than `coveragepy`'s implementation:

https://github.com/googleapis/python-storage/blob/main/setup.py#L45-L48

* * *

Also, in regard to release-please expecting a structure similar to `googleapis/python-storage`: they use `google/cloud/storage/version.py`, we use `butter_cms/__version__.py`. That doesn't look similar, are we sure it's 100% compatible?

Also, could you point me to the place in the docs/code/wherever that explains that? [Place that you've linked](https://github.com/googleapis/release-please?tab=readme-ov-file#strategy-language-types-supported) tells about:

> [setup.py, setup.cfg, CHANGELOG.md](https://github.com/googleapis/python-storage) and optionally a pyproject.toml and a <project>/__init__.py

But not about `version.py`, so I'm curious where you were able to find that info.

Reply:
That is an interesting point, I could argue that executing a file is more overkill 😄  but you're right, we have a control over it, so it does not pose a security threat. I'll rewrite it similarly to python-storage approach.

Regarding the `__version__.py` vs `version.py`. Here is code of that "python strategy" in release-please: https://github.com/googleapis/release-please/blob/main/src/strategies/python.ts - it looks for version.py. I tested this in my fork and automatic bumps are done only if we have version.py - this is fixed in #8.

Also regarding the most simple solution - having it defined in setup.py - this is indeed the most simple approach, but we might need to access it down the line. I think it's also a good practice to keep it separated.

Reply:
> Here is code of that "python strategy" in release-please: https://github.com/googleapis/release-please/blob/main/src/strategies/python.ts

Thanks! I had a gut feeling that I should look for this in the code, but wasn't sure where exactly, so that was helpful.

> it looks for version.py. I tested this in my fork and automatic bumps are done only if we have version.py - this is fixed in https://github.com/ButterCMS/buttercms-python/pull/8

Great!

* * *

That was a productive discussion, thanks :)